apps/api/src/domain/models/resource.entity.ts

```diff
@@ -1,79 +1,39 @@
-import { InternalError, isInternalError } from '../../lib/errors/InternalError';
-import cloneToPlainObject from '../../lib/utilities/cloneToPlainObject';
+import { InternalError } from '../../lib/errors/InternalError';
 import capitalizeFirstLetter from '../../lib/utilities/strings/capitalizeFirstLetter';
 import { DeepPartial } from '../../types/DeepPartial';
 import { DTO } from '../../types/DTO';
 import { ResultOrError } from '../../types/ResultOrError';
-import formatResourceCompositeIdentifier from '../../view-models/presentation/formatResourceCompositeIdentifier';
+import formatResourceCompositeIdentifier from '../../view-models/presentation/formatAggregateCompositeIdentifier';
 import DisallowedContextTypeForResourceError from '../domainModelValidators/errors/context/invalidContextStateErrors/DisallowedContextTypeForResourceError';
 import { Valid } from '../domainModelValidators/Valid';
-import { AggregateId } from '../types/AggregateId';
 import { HasAggregateId } from '../types/HasAggregateId';
 import { ResourceCompositeIdentifier } from '../types/ResourceCompositeIdentifier';
 import { ResourceType } from '../types/ResourceType';
+import { Aggregate } from './aggregate.entity';
 import { getAllowedContextsForModel } from './allowedContexts/isContextAllowedForGivenResourceType';
-import BaseDomainModel from './BaseDomainModel';
```

Comment:
Is `BaseDomainModel` still relevant to our code?  And how does this shift the separation we were talking about with the create versus update decision tree?

Reply:
`BaseDomainModel` adds `clone` and `toDTO` (see `BaseDomainModel.ts`). If mixins were cleaner, I'd use `Cloneable` and `DTOable` mixins to do this instead. I don't think there's a whole lot of baggage at this point because we know that any `Aggregate` and any `Resource` will need `toDTO` and `clone`.